# Mini Project 5-5 Explore Hypothesis Testing

## Introduction

You work for an environmental think tank called Repair Our Air (ROA). ROA is formulating policy recommendations to improve the air quality in America, using the Environmental Protection Agency's Air Quality Index (AQI) to guide their decision making. An AQI value close to 0 signals "little to no" public health concern, while higher values are associated with increased risk to public health. 

They've tasked you with leveraging AQI data to help them prioritize their strategy for improving air quality in America.

ROA is considering the following decisions. For each, construct a hypothesis test and an accompanying visualization, using your results of that test to make a recommendation:

1. ROA is considering a metropolitan-focused approach. Within California, they want to know if the mean AQI in Los Angeles County is statistically different from the rest of California.
2. With limited resources, ROA has to choose between New York and Ohio for their next regional office. Does New York have a lower AQI than Ohio?
3. A new policy will affect those states with a mean AQI of 10 or greater. Will Michigan be affected by this new policy?

**Notes:**
1. For your analysis, you'll default to a 5% level of significance.
2. Throughout the lab, for two-sample t-tests, use Welch's t-test (i.e., setting the `equal_var` parameter to `False` in `scipy.stats.ttest_ind()`). This will account for the possibly unequal variances between the two groups in the comparison.

## Step 1: Imports

To proceed with your analysis, import `pandas` and `numpy`. To conduct your hypothesis testing, import `stats` from `scipy`.

#### Import Packages

In [2]:
# Import relevant packages
import pandas as pd
import numpy as np
from scipy import stats

You are also provided with a dataset with national Air Quality Index (AQI) measurements by state over time for this analysis. `Pandas` was used to import the file `c4_epa_air_quality.csv` as a dataframe named `aqi`. As shown in this cell, the dataset has been automatically loaded in for you. You do not need to download the .csv file, or provide more code, in order to access the dataset and proceed with this lab. Please continue with this activity by completing the following instructions.

**Note:** For purposes of your analysis, you can assume this data is randomly sampled from a larger population.

#### Load Dataset

In [16]:
# IMPORT YOUR DATA
aqi = pd.read_csv('c4_epa_air_quality (1).csv', index_col=0)
aqi.shape

(260, 9)

## Step 2: Data Exploration

### Before proceeding to your deliverables, explore your datasets.

Use the following space to surface descriptive statistics about your data. In particular, explore whether you believe the research questions you were given are readily answerable with this data.

In [18]:
# Use head() to show a sample of data
aqi.head()

,date_local,state_name,county_name,city_name,local_site_name,parameter_name,units_of_measure,arithmetic_mean,aqi
0,01-01-2018,Arizona,Maricopa,Buckeye,BUCKEYE,Carbon monoxide,Parts per million,0.473684,7
1,01-01-2018,Ohio,Belmont,Shadyside,Shadyside,Carbon monoxide,Parts per million,0.263158,5
2,01-01-2018,Wyoming,Teton,Not in a city,Yellowstone National Park - Old Faithful Snow ...,Carbon monoxide,Parts per million,0.111111,2
3,01-01-2018,Pennsylvania,Philadelphia,Philadelphia,North East Waste (NEW),Carbon monoxide,Parts per million,0.300000,3
4,01-01-2018,Iowa,Polk,Des Moines,CARPENTER,Carbon monoxide,Parts per million,0.215789,3


In [10]:
# check varibles
aqi.columns 

Index(['date_local', 'state_name', 'county_name', 'city_name',
       'local_site_name', 'parameter_name', 'units_of_measure',
       'arithmetic_mean', 'aqi'],
      dtype='object')

In [14]:
# Use describe() to summarize AQI
aqi.describe()

,arithmetic_mean,aqi
count,260.000000,260.000000
mean,0.403169,6.757692
std,0.317902,7.061707
min,0.000000,0.000000
25%,0.200000,2.000000
50%,0.276315,5.000000
75%,0.516009,9.000000
max,1.921053,50.000000


In [15]:
# For a more thorough examination of observations by state use values_counts()
state_counts = aqi['state_name'].value_counts()
state_counts


state_name
California              66
Arizona                 14
Ohio                    12
Florida                 12
Texas                   10
Pennsylvania            10
New York                10
Colorado                 9
Michigan                 9
Minnesota                7
New Jersey               6
Indiana                  5
Nevada                   4
Maryland                 4
Massachusetts            4
Oklahoma                 4
North Carolina           4
Virginia                 4
Connecticut              4
Wyoming                  3
Utah                     3
Hawaii                   3
Illinois                 3
Missouri                 3
Iowa                     3
Kentucky                 3
Vermont                  3
Rhode Island             2
Alaska                   2
Georgia                  2
Washington               2
Idaho                    2
Nebraska                 2
Montana                  2
Tennessee                2
Maine                    2
New Mexico       

#### **Question 1: From the preceding data exploration, what do you recognize?**

A: from the above data exploration, we get an idea about the columns, mean, median, mode and percentiles through describe and also the value counts of each state
also it is possible to explore the research questions that follows in this notebook are answerable with this data.

## Step 3. Statistical Tests

Before you proceed, recall the following steps for conducting hypothesis testing:

1. Formulate the null hypothesis and the alternative hypothesis.<br>
2. Set the significance level.<br>
3. Determine the appropriate test procedure.<br>
4. Compute the p-value.<br>
5. Draw your conclusion.

### Hypothesis 1: ROA is considering a metropolitan-focused approach. Within California, they want to know if the mean AQI in Los Angeles County is statistically different from the rest of California.

Before proceeding with your analysis, it will be helpful to subset the data for your comparison.

In [24]:
# Create dataframes for each sample being compared in your test
California = aqi[aqi['state_name'] == "California"]  
California

la = aqi[aqi['county_name'] == "Los Angeles"] 
la 

###### California variable has all the county's and la variable stores only Los Angeles county data

,date_local,state_name,county_name,city_name,local_site_name,parameter_name,units_of_measure,arithmetic_mean,aqi
33,01-01-2018,California,Los Angeles,Lancaster,Lancaster-Division Street,Carbon monoxide,Parts per million,0.394737,7
42,01-01-2018,California,Los Angeles,Santa Clarita,Santa Clarita,Carbon monoxide,Parts per million,0.394737,7
61,01-01-2018,California,Los Angeles,Pasadena,Pasadena,Carbon monoxide,Parts per million,0.789474,16
76,01-01-2018,California,Los Angeles,Los Angeles,LAX Hastings,Carbon monoxide,Parts per million,0.863158,17
109,01-01-2018,California,Los Angeles,Los Angeles,Los Angeles-North Main Street,Carbon monoxide,Parts per million,0.994737,17
110,01-01-2018,California,Los Angeles,Los Angeles,Los Angeles-North Main Street,Carbon monoxide,Parts per million,0.900000,16
119,01-01-2018,California,Los Angeles,Reseda,Reseda,Carbon monoxide,Parts per million,1.015789,19
132,01-01-2018,California,Los Angeles,Compton,Compton,Carbon monoxide,Parts per million,1.742105,40
163,01-01-2018,California,Los Angeles,Azusa,Azusa,Carbon monoxide,Parts per million,0.673684,10
172,01-01-2018,California,Los Angeles,Pico Rivera,Pico Rivera #2,Carbon monoxide,Parts per million,1.047368,18


In [25]:
# Check head
California.head()

,date_local,state_name,county_name,city_name,local_site_name,parameter_name,units_of_measure,arithmetic_mean,aqi
16,01-01-2018,California,San Bernardino,Ontario,Ontario Near Road (Etiwanda),Carbon monoxide,Parts per million,0.747368,11
18,01-01-2018,California,Sacramento,Arden-Arcade,Sacramento-Del Paso Manor,Carbon monoxide,Parts per million,0.752632,16
26,01-01-2018,California,Orange,La Habra,La Habra,Carbon monoxide,Parts per million,0.673684,13
27,01-01-2018,California,Alameda,Not in a city,Berkeley- Aquatic Park,Carbon monoxide,Parts per million,1.088889,15
33,01-01-2018,California,Los Angeles,Lancaster,Lancaster-Division Street,Carbon monoxide,Parts per million,0.394737,7


In [26]:
la.head()

,date_local,state_name,county_name,city_name,local_site_name,parameter_name,units_of_measure,arithmetic_mean,aqi
33,01-01-2018,California,Los Angeles,Lancaster,Lancaster-Division Street,Carbon monoxide,Parts per million,0.394737,7
42,01-01-2018,California,Los Angeles,Santa Clarita,Santa Clarita,Carbon monoxide,Parts per million,0.394737,7
61,01-01-2018,California,Los Angeles,Pasadena,Pasadena,Carbon monoxide,Parts per million,0.789474,16
76,01-01-2018,California,Los Angeles,Los Angeles,LAX Hastings,Carbon monoxide,Parts per million,0.863158,17
109,01-01-2018,California,Los Angeles,Los Angeles,Los Angeles-North Main Street,Carbon monoxide,Parts per million,0.994737,17


#### Formulate your hypothesis:

In [28]:
sampled_california = California.sample(n=20, replace = True, random_state = 42)
sampled_california.info() 

<class 'pandas.core.frame.DataFrame'>
Index: 20 entries, 203 to 224
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   date_local        20 non-null     object 
 1   state_name        20 non-null     object 
 2   county_name       20 non-null     object 
 3   city_name         20 non-null     object 
 4   local_site_name   20 non-null     object 
 5   parameter_name    20 non-null     object 
 6   units_of_measure  20 non-null     object 
 7   arithmetic_mean   20 non-null     float64
 8   aqi               20 non-null     int64  
dtypes: float64(1), int64(1), object(7)
memory usage: 1.6+ KB


In [30]:
sampled_california['aqi'].mean()

np.float64(11.8)

In [31]:
sampled_la = la.sample(n=20, replace = True, random_state = 42)
sampled_la.info()

<class 'pandas.core.frame.DataFrame'>
Index: 20 entries, 119 to 110
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   date_local        20 non-null     object 
 1   state_name        20 non-null     object 
 2   county_name       20 non-null     object 
 3   city_name         20 non-null     object 
 4   local_site_name   20 non-null     object 
 5   parameter_name    20 non-null     object 
 6   units_of_measure  20 non-null     object 
 7   arithmetic_mean   20 non-null     float64
 8   aqi               20 non-null     int64  
dtypes: float64(1), int64(1), object(7)
memory usage: 1.6+ KB


In [32]:
sampled_la['aqi'].mean()

np.float64(21.75)

**Formulate your null and alternative hypotheses:**

*   $H_0$: There is no difference in the mean AQI between Los Angeles County and the rest of California.
*   $H_A$: There is a difference in the mean AQI between Los Angeles County and the rest of California.


#### Set the significance level:

In [35]:
# For this analysis, the significance level is 5%, that is 0.05

#### Determine the appropriate test procedure:

Here, you are comparing the sample means between two independent samples. Therefore, you will utilize a **two-sample  𝑡-test**.

#### Compute the P-value

In [36]:
# Compute your p-value here
# n < 30, t-distribution: sample a, sample b, assume not same variance.

stats.ttest_ind(a=sampled_california['aqi'], b=sampled_la['aqi'], equal_var=False) 

TtestResult(statistic=np.float64(-3.1678167798999386), pvalue=np.float64(0.003046410459439079), df=np.float64(37.592028570831275))

#### **Question 2. What is your P-value for hypothesis 1, and what does this indicate for your null hypothesis?**

In [37]:
# Extracting pvalue and make the test
statistic, pvalue = stats.ttest_ind(a=sampled_california['aqi'], b=sampled_la['aqi'], equal_var=False)

print ("pvalue:",pvalue)
print()

if pvalue < 0.05:
    
    print('pvalue < 0.05, Reject Ho.')          # Ha: There is a difference in the mean
else:
    print('pvalue > 0.05, Fail to reject Ho.')  # Ho: There is no difference in the mean

pvalue: 0.003046410459439079

pvalue < 0.05, Reject Ho.


A: there is a statistically significant difference between the mean aqi values of the two under comparison.

### Hypothesis 2: With limited resources, ROA has to choose between New York and Ohio for their next regional office. Does New York have a lower AQI than Ohio?

Before proceeding with your analysis, it will be helpful to subset the data for your comparison.

In [39]:
# Create dataframes for each sample being compared in your test
NY = aqi[aqi['state_name'] == "New York"]
NY

Ohio = aqi[aqi['state_name'] == "Ohio"]
Ohio

,date_local,state_name,county_name,city_name,local_site_name,parameter_name,units_of_measure,arithmetic_mean,aqi
1,01-01-2018,Ohio,Belmont,Shadyside,Shadyside,Carbon monoxide,Parts per million,0.263158,5
12,01-01-2018,Ohio,Hamilton,Cincinnati,Taft NCore,Carbon monoxide,Parts per million,0.252632,3
22,01-01-2018,Ohio,Stark,Canton,Canton,Carbon monoxide,Parts per million,0.394737,6
51,01-01-2018,Ohio,Summit,Akron,NIHF STEM MS,Carbon monoxide,Parts per million,0.083333,3
59,01-01-2018,Ohio,Cuyahoga,Cleveland,GT Craig NCore,Carbon monoxide,Parts per million,0.250000,3
120,01-01-2018,Ohio,Cuyahoga,Cleveland,Galleria,Carbon monoxide,Parts per million,0.273684,3
149,01-01-2018,Ohio,Franklin,Columbus,Morse Rd,Carbon monoxide,Parts per million,0.184211,3
191,01-01-2018,Ohio,Franklin,Columbus,Smoky Row Near Road,Carbon monoxide,Parts per million,0.115789,2
215,01-01-2018,Ohio,Cuyahoga,Warrensville Heights,Cleveland Near Road,Carbon monoxide,Parts per million,0.321053,5
231,01-01-2018,Ohio,Montgomery,Dayton,Reibold,Carbon monoxide,Parts per million,0.163158,2


In [41]:
# Check head
NY.head()

,date_local,state_name,county_name,city_name,local_site_name,parameter_name,units_of_measure,arithmetic_mean,aqi
90,01-01-2018,New York,Erie,Cheektowaga,Buffalo Near-Road,Carbon monoxide,Parts per million,0.252632,3
113,01-01-2018,New York,Bronx,New York,PFIZER LAB SITE,Carbon monoxide,Parts per million,0.289474,3
124,01-01-2018,New York,Monroe,Rochester,ROCHESTER 2,Carbon monoxide,Parts per million,0.200000,2
167,01-01-2018,New York,New York,New York,CCNY,Carbon monoxide,Parts per million,0.200000,2
173,01-01-2018,New York,Queens,New York,Queens College Near Road,Carbon monoxide,Parts per million,0.273684,3


In [42]:
Ohio.head()

,date_local,state_name,county_name,city_name,local_site_name,parameter_name,units_of_measure,arithmetic_mean,aqi
1,01-01-2018,Ohio,Belmont,Shadyside,Shadyside,Carbon monoxide,Parts per million,0.263158,5
12,01-01-2018,Ohio,Hamilton,Cincinnati,Taft NCore,Carbon monoxide,Parts per million,0.252632,3
22,01-01-2018,Ohio,Stark,Canton,Canton,Carbon monoxide,Parts per million,0.394737,6
51,01-01-2018,Ohio,Summit,Akron,NIHF STEM MS,Carbon monoxide,Parts per million,0.083333,3
59,01-01-2018,Ohio,Cuyahoga,Cleveland,GT Craig NCore,Carbon monoxide,Parts per million,0.250000,3


In [43]:
sampled_NY = NY.sample(n=20, replace = True, random_state = 42)
sampled_NY.info()

<class 'pandas.core.frame.DataFrame'>
Index: 20 entries, 184 to 113
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   date_local        20 non-null     object 
 1   state_name        20 non-null     object 
 2   county_name       20 non-null     object 
 3   city_name         20 non-null     object 
 4   local_site_name   20 non-null     object 
 5   parameter_name    20 non-null     object 
 6   units_of_measure  20 non-null     object 
 7   arithmetic_mean   20 non-null     float64
 8   aqi               20 non-null     int64  
dtypes: float64(1), int64(1), object(7)
memory usage: 1.6+ KB


In [44]:
sampled_NY['aqi'].mean()

np.float64(2.55)

In [46]:
sampled_Ohio = Ohio.sample(n=20, replace = True, random_state = 42)
sampled_Ohio.info()

<class 'pandas.core.frame.DataFrame'>
Index: 20 entries, 149 to 12
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   date_local        20 non-null     object 
 1   state_name        20 non-null     object 
 2   county_name       20 non-null     object 
 3   city_name         20 non-null     object 
 4   local_site_name   20 non-null     object 
 5   parameter_name    20 non-null     object 
 6   units_of_measure  20 non-null     object 
 7   arithmetic_mean   20 non-null     float64
 8   aqi               20 non-null     int64  
dtypes: float64(1), int64(1), object(7)
memory usage: 1.6+ KB


In [47]:
sampled_Ohio['aqi'].mean()

np.float64(2.9)

**Formulate your null and alternative hypotheses:**

*   $H_0$: The mean AQI of New York is greater than or equal to that of Ohio.
*   $H_A$: The mean AQI of New York is **below** that of Ohio.


#### Significance Level (remains at 5%)

#### Determine the appropriate test procedure:

Here, you are comparing the sample means between two independent samples in one direction. Therefore, you will utilize a **two-sample  𝑡-test**.

#### Compute the P-value

In [48]:
# Compute your p-value here
stats.ttest_ind(a=sampled_NY['aqi'], b=sampled_Ohio['aqi'], equal_var=False) 

TtestResult(statistic=np.float64(-1.230374389448749), pvalue=np.float64(0.22955971531662292), df=np.float64(26.031852117497472))

#### **Question 3. What is your P-value for hypothesis 2, and what does this indicate for your null hypothesis?**

In [49]:
# Your code here.
statistic, pvalue = stats.ttest_ind(a=sampled_NY['aqi'], b=sampled_Ohio['aqi'], equal_var=False)

print ("pvalue:",pvalue)
print()

if pvalue < 0.05:
    
    print('pvalue < 0.05, Reject Ho.')          # Ha: There is a difference in the mean
else:
    print('pvalue > 0.05, Fail to reject Ho.')  # Ho: There is no difference in the mean

pvalue: 0.22955971531662292

pvalue > 0.05, Fail to reject Ho.


A: There is no statistically significant difference between the mean aqi values of the two states under comparison.

###  Hypothesis 3: A new policy will affect those states with a mean AQI of 10 or greater. Will Michigan be affected by this new policy?

Before proceeding with your analysis, it will be helpful to subset the data for your comparison.

In [55]:
# Create dataframes for each sample being compared in your test
michigan_sample = aqi[aqi['state_name'] == 'Michigan']['aqi']
michigan_sample

65      5
122     8
123     9
129    11
192    13
207    10
226     2
242     7
248     8
Name: aqi, dtype: int64

In [56]:
# Hypothesized mean AQI (threshold)
pop_mean = 10

**Formulate your null and alternative hypotheses here:**

*   $H_0$: The mean AQI of Michigan is less than or equal to 10.
*   $H_A$: The mean AQI of Michigan is greater than 10.


#### Significance Level (remains at 5%)

#### Determine the appropriate test procedure:

Here, you are comparing one sample mean relative to a particular value in one direction. Therefore, you will utilize a **one-sample  𝑡-test**. 

#### Compute the P-value

In [63]:
# One-sample t-test
t_stat, p_value_two_tailed = stats.ttest_1samp(michigan_sample, pop_mean)

In [61]:
# Convert to one-tailed p-value (testing if mean > 10)
if t_stat > 0:
    p_value_one_tailed = p_value_two_tailed / 2
else:
    p_value_one_tailed = 1 - (p_value_two_tailed / 2)

In [ ]:
# Output
print("Mean AQI (Michigan):", michigan_sample.mean())
print("Test statistic:", t_stat)
print("One-tailed p-value:", p_value_one_tailed)

# result
alpha = 0.05
if p_value_one_tailed < alpha:
    print("Michigan will likely be affected. The mean AQI is significantly greater than 10.")
else:
    print("Michigan will likely not be affected. The data does not support a mean AQI above 10.")

Mean AQI (Michigan): 8.11111111111111
Test statistic: -1.7395913343286131
One-tailed p-value: 0.9399405193140109
Michigan will likely not be affected. The data does not support a mean AQI above 10.


#### **Question 4. What is your P-value for hypothesis 3, and what does this indicate for your null hypothesis?**

A: Since the p-value is 0.94 and is much greater than 0.05, we fail to reject the null hypothesis.

## Step 4. Results and Evaluation

Now that you've completed your statistical tests, you can consider your hypotheses and the results you gathered.

#### **Question 5. Did your results show that the AQI in Los Angeles County was statistically different from the rest of California?**

A:yes. there is a statistically significant difference between the mean aqi values between the LA county and California state

#### **Question 6. Did New York or Ohio have a lower AQI?**

A: so failing to reject null hypothesis means the data isn't strong enough to say New York has a lower aqi than Ohio. But it doesn't prove the opposite either.

#### **Question 7: Will Michigan be affected by the new policy impacting states with a mean AQI of 10 or greater?**



A: Michigan will not be affected by the new policy, as the data does not support that its average aqi meets or exceeds the threshold of 10.

# Conclusion

**What are key takeaways from this project?**

A: learnt the use of hypothesis testing and it's importance to prove anything via statistical testing

**What would you consider presenting to your manager as part of your findings?**

A:  the results of the hypothesis testing along with a graph


**What would you convey to external readers?**

A: the end results and evaluation to the external readers. 
